cli/cmd/root.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/util/homedir"
+)
+
+// NewDefaultRootCmd creates a new root command as entrypoint for all subcommands with the default
+// subcommands constructors.
+func NewDefaultRootCmd() *cobra.Command {
+	return NewRootCmd(
+		NewDefaultMarketplaceCmd(),
+		NewDefaultProvisionCmd(),
+		NewDefaultDeprovisionCmd(),
+		NewDefaultInstancesCmd(),
+		NewDefaultBindCmd(),
+		NewDefaultUnbindCmd(),
+		NewDefaultCredentialsCmd(),
+	)
+}
+
+// NewRootCmd creates a new root command as entrypoint for all subcommands.
+func NewRootCmd(
+	marketplaceCmd *cobra.Command,
+	provisionCmd *cobra.Command,
+	deprovisionCmd *cobra.Command,
+	instancesCmd *cobra.Command,
+	bindCmd *cobra.Command,
+	unbindCmd *cobra.Command,
+	credentialsCmd *cobra.Command,
+) *cobra.Command {
+	var kubeconfig string
+
+	cmd := &cobra.Command{
+		Use: "metabrokerctl",
```

Comment:
I don't think this is necessary? (At least, for the root command; having less code just generally seems better, even if it's just one line.)

Reply:
This is not _necessary_ but then the help command shows empty, which is weird. Well, ok, it's wrong.